packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -49,7 +52,11 @@ def supportsNamespace() {
 
 android {
   if (supportsNamespace()) {
+<% if (project.nitro) { -%>
+    namespace "com.margelo.nitro.<%- project.package -%>"
```

Comment:
isn't this the library's namespace? then it shouldn't have margelo.com since that's not the library's organization.

Reply:
hmm this doesn't seem good. i wouldn't feel comfortable having my own library under margelo's domain.

Reply:
Yea Nitro expects margelo.com namespaces for all Hybrid Objects.
This was to simplify the ProGuard annotations (if something is missing from com.margelo it was stripped out, which we explicitly deal with) - and also it makes cross-referencing Hybrid Objects from other Nitro Modules easier since we don't have to worry about different namespaces here.
Also, it's at least a little bit of branding 😄

Reply:
> i wouldn't feel comfortable having my own library under margelo's domain.

All the generated code from Turbo Modules are under facebook's domain. This is the same here, the generated code is under margelo domain.

Nitro expects registered hybrid objects to be under the same domain as well, not any custom configurable domain - which is the same as how it works in C++ Turbo Modules.
 This was just to simplify the whole thing.

Reply:
> All the generated code from Turbo Modules are under facebook's domain. This is the same here, the generated code is under margelo domain.

Having the generated code or a specific part of the code vs the entire package like it's done here are quite different things. I'm sure bigger companies would have restrictions on using their namespace and publishing to their internal registries.

Reply:
yea I understand, but this is not gonna change anytime soon as it might require larger changes in the Nitrogen codebase.

Reply:
If a bigger company reaches out to me because they want to use Nitro but can't because of namespace restrictions, then I'll work on that with them. For now, I don't see a problem in it tbh.